src/pytest_mock_resources/fixture/database/relational/generic.py

```diff
@@ -154,6 +155,59 @@ def manage(self, session=None):
         finally:
             self.engine.dispose()
 
+    def manage_sync(self, session=None):
+        try:
+            self._run_actions()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    session_factory = session
+                else:
+                    session_factory = sessionmaker(bind=self.engine)
+
+                Session = scoped_session(session_factory)
+                session = Session(bind=self.engine)
+                yield session
+                session.close()
+            else:
+                yield self.engine
+        finally:
+            self.engine.dispose()
+
+    def manage_async(self, session=None):
+        try:
+            self._run_actions()
+
+            engine = self._get_async_engine()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    yield session
+                else:
+                    yield sessionmaker(
+                        engine, expire_on_commit=False, class_=AsyncSession
+                    )
+            else:
+                yield engine
```

Comment:
similarly, i think `engine` also needs to be (presumably `await`) `engine.dispose()`'d similarly to the `self.engine`

Reply:
Unlike the synchronous engine, the asynchronous one does not need to be disposed of since no connection is initialized until `connect` or `begin` is called explicitly. That's why the docs suggest the use of a context manager like so:
```
 async with engine.begin() as conn:
    ...
```